pkg/parser/tree.go

```diff
@@ -32,7 +32,9 @@ var (
 
 var (
 	validCreIdRegex    = regexp.MustCompile(`^[A-Za-z0-9-]+$`)
-	validBase58IdRegex = regexp.MustCompile(`^[1-9A-HJ-NP-Za-km-z-]+$`)
+	validBase58IdRegex = regexp.MustCompile(`^[1-9A-Za-z]+$`)
+	minRuleIdLength    = 12
```

Comment:
This is fine, but it can also be included in the regex with a limit on the character class.

^[A-Za-z0-9-]{12,}$